url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/parser/MethodCommand.java

```diff
@@ -0,0 +1,124 @@
+ * http://www.sdi-consulting.be
+ * mailto://info@sdi-consulting.be
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.sdicons.scripty.parser;
+
+import com.sdicons.scripty.cmdlib.CmdUtil;
+import com.sdicons.scripty.parser.CommandException;
+import com.sdicons.scripty.parser.ICommand;
+import com.sdicons.scripty.parser.IContext;
+import com.sdicons.scripty.parser.IEval;
+import com.sdicons.scripty.spec.args.ArgSpecException;
+import com.sdicons.scripty.spec.args.IArgList;
+import com.sdicons.scripty.spec.map.ArgListMapping;
+import com.sdicons.scripty.spec.map.ArgMappingException;
+import com.sdicons.scripty.spec.map.IResultMapping;
+import com.sdicons.scripty.spec.map.ResultMappingException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class MethodCommand
+implements ICommand
+{
+    private IArgList argList;
+    private Object instance;
+    private ArgListMapping argMapping;
+    private IResultMapping resultMapping;
+    private Method method;
+
+    public MethodCommand(Method aMethod)
+    {
+        argList = null;
+        method = aMethod;
+        instance = null;
+        argMapping = null;
+        resultMapping = null;
+    }
+
+    public MethodCommand(Object aInstance, Method aMethod, IArgList aArgList, ArgListMapping aArgListMapping, IResultMapping aResultMapping)
+    {
+        argList = aArgList;
+        method = aMethod;
+        instance = aInstance;
+        argMapping = aArgListMapping;
+        resultMapping = aResultMapping;
+    }
+
+    public Object execute(IEval aEval, IContext aCtx, Object[] aArgs)
+    throws CommandException
+    {
+        try
+        {
+            Object[] lArgs = aArgs;
+
+            if (argList != null)
+            {
+                lArgs = argList.guard(aArgs, aCtx);
+            }
+            
+            if(argMapping != null)
+            {
+                lArgs = argMapping.map(aEval, aCtx, lArgs);
+            }
+            
+            Object lResult = method.invoke(instance, lArgs);
+
+            if(resultMapping != null)
+            {
+                resultMapping.map(lResult, aCtx);
+            }
+
+            return lResult;
+        }
+        catch (IllegalAccessException e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 96, 104, 108, 113, 119.